xdem/coreg.py

```diff
@@ -468,6 +469,20 @@ def fit(self, reference_dem: Union[np.ndarray, np.ma.masked_array],
         if weights is not None:
             raise NotImplementedError("Weights have not yet been implemented")
 
+        # If both DEMs are Rasters, validate that 'dem_to_be_aligned' is in the right grid. Then extract its data.
+        if isinstance(dem_to_be_aligned, gu.Raster) and isinstance(reference_dem, gu.Raster):
+            dem_to_be_aligned = dem_to_be_aligned.reproject(reference_dem, silent=True).data
```

Comment:
after seeing so many `silent=True`, I'm wondering if we'd not prefer `silent=True` by default on `Raster.reproject` as well!

Reply:
A difference is that `Raster.reproject(..., silent=True)` may actually have some unforeseen consequences: it returns a copy, so any action after that will propagate to the original raster too.

While I'm not perfectly happy with the warning that shows up here and there, it would almost have to be replaced by another protection mechanism to be completely safe....

Reply:
True, and it's not everyday we reproject on same CRS.
Good as is!